maps/_map.dm

```diff
@@ -146,6 +146,18 @@ proc/get_base_turf(var/z)
 	var/datum/zLevel/L = map.zLevels[z]
 	return L.base_turf
 
+proc/change_base_turf(var/choice,var/new_base_path,var/update_old_base = 0)
+	if(update_old_base)
+		for(var/turf/T in turfs)

```

Comment:
This is a really large loop, hence why it freezes everything for almost a minute. Is there no list that loops through a particular Z-level's contents ?